staking/keeper.go

```diff
@@ -27,8 +28,12 @@ var (
 	StakingSequenceKey        = []byte{0x24} // prefix for each key for staking sequence map
 )
 
-type AckRetriever interface {
+// ModuleCommunicator manages different module interaction
+type ModuleCommunicator interface {
```

Comment:
Please elaborate on the purpose of this module, how do we decide what goes in here etc